lib/data_provider.py

```diff
@@ -124,9 +128,11 @@ def get_simple_mixed_batch(
 
     # These operations merge the datasets in a way that intersperses
     # elements from each, rather than just concatenating them.
-    if labeled_dataset_name == "imagenet_32":
+    if labeled_dataset_name == "imagenet_32" or mode == "labeled":
         # Don't waste batch space for imagenet pre-training.
         dataset = labeled_dataset
+    elif mode == "unlabeled":
```

Comment:
Please also remove this clause here, per my other comments -- it doesn't make sense to train a classifier with only unlabeled data.